src/utils.ts

```diff
@@ -248,7 +254,7 @@ export function executeLegacyOperation(
       const optionsIndex = args.length - 2;
       args[optionsIndex] = Object.assign({}, args[optionsIndex], { session: session });
     } else if (opOptions.session && opOptions.session.hasEnded) {
-      throw new MongoDriverError('Use of expired sessions is not permitted');
+      throw new MongoInvalidArgumentError('Use of expired sessions is not permitted');
```

Comment:
I'm a little torn about this one. The same error message appears in `src/operations/execute_operation.ts:88` as well so I feel like the verdict has to be consistent for the two cases. That being said I'm not sure whether `MongoInvalidArgumentError` or `MongoResourceClosedError` would fit better here. Definitely open to other perspectives.

Reply:
I had left the error in `src/operations/execute_operation.ts:88` because of that ambiguity. Was intending to leave that until we added the `MongoResourceClosedError` and its children, so yeah, I think you're right about changing this to `MongoResourceClosedError` as well